ApiBundle/Controller/NodeController.php

```diff
@@ -66,11 +68,18 @@ public function showOrCreateAction(Request $request, $nodeId, $errorNode)
         $language = $request->get('language', $currentSiteDefaultLanguage);
         $siteId = $currentSiteManager->getCurrentSiteId();
         $node = $this->findOneNode($nodeId, $language, $siteId, $request->get('version'));
-
+        if (!$errorNode && $node) {
```

Comment:
I tried your solution and it's not making the code clearer in my sense cause we have to pass many parameters to each method.